core/src/main/scala/io/github/gaelrenoux/tranzactio/ConnectionSource.scala

```diff
@@ -39,6 +46,16 @@ object ConnectionSource {
       }
     }
 
+    def runTransactionStream[R, E, A](task: Connection => ZStream[R, E, A], commitOnFailure: => Boolean = false)
+       (implicit errorStrategies: ErrorStrategiesRef, trace: Trace): ZStream[R, Either[DbException, E], A] = {
+
+      ZStream.acquireReleaseWith(openConnection.tap(c => setAutoCommit(c, autoCommit = false)).mapError(Left(_)))
```

Comment:
By setting the auto-commit in a `tap` in the acquisition step (instead of doing it later, as is done in `runTransaction`), you're introducing a potential issue: if setting the auto-commit fails, the connection won't get closed, as the acquisition step won't have ended with a success.